File: foxy/parent/src/command/vanilla/entertainment/memes/LaranjoExecutor.ts
import { ExecutorParams } from "../../../structures/CommandExecutor";
import { bot } from "../../../../FoxyLauncher";

export default class LaranjoExecutor {
    async execute({ context, endCommand }: ExecutorParams) {
        const content = context.getOption<string>("text", false);
        context.sendDefer();

        const laranjoImage = await bot.rest.foxy.getArtistryImage("/memes/laranjo", {
            text: content
        });

        const file = new File([laranjoImage], "laranja_laranjo.png", { type: "image/png" });

        context.reply({
            file: {
                name: "laranja_laranjo.png",
                blob: file
            }
        });
        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/entertainment/media/LaranjoExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.entertainment.media

import io.ktor.client.call.*
import io.ktor.utils.io.*
import kotlinx.coroutines.withContext
import kotlinx.io.readByteArray
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put
import net.cakeyfox.foxy.interactions.FoxyInteractionContext
import net.cakeyfox.foxy.interactions.commands.FoxySlashCommandExecutor
import net.dv8tion.jda.api.utils.FileUpload

class LaranjoExecutor : FoxySlashCommandExecutor() {
    override suspend fun execute(context: FoxyInteractionContext) {
        context.defer()
        val text = context.getOption<String>("text")!!

        val laranjoImage = withContext(context.foxy.coroutineDispatcher) {
            context.foxy.artistryClient.generateImage("memes/laranjo", buildJsonObject {
                put("text", text)
            })
        }

        if (laranjoImage.status.value !in 200..299) {
            throw IllegalArgumentException("Error while generating image! Received ${laranjoImage.status}")
        }

        val image = laranjoImage.body<ByteReadChannel>().readRemaining().readByteArray()
        val file = FileUpload.fromData(image, "laranjo_${context.user.id}.png")

        context.reply {
            files.plusAssign(file)
        }
    }
}